mesonbuild/backend/backends.py

```diff
@@ -113,12 +113,13 @@ def __init__(self, source_dir: str, build_dir: str, prefix: str,
         self.version = version
 
 class TargetInstallData:
-    def __init__(self, fname: str, outdir: str, aliases: T.Dict[str, str], strip: bool,
-                 install_name_mappings: T.Dict, rpath_dirs_to_remove: T.Set[bytes],
+    def __init__(self, fname: str, outdir: str, outdir_name: str, aliases: T.Dict[str, str],
+                 strip: bool, install_name_mappings: T.Dict, rpath_dirs_to_remove: T.Set[bytes],
                  install_rpath: str, install_mode: 'FileMode', subproject: str,
                  optional: bool = False, tag: T.Optional[str] = None):
         self.fname = fname
         self.outdir = outdir
+        self.out_name = os.path.join(outdir_name, fname)
```

Comment:
```suggestion
        self.out_name = os.path.join(outdir_name, os.path.basename(fname))
```

Please update the test to verify that files built from a subdir/meson.build are correctly handled (that is how I discovered this edge case).